src/index.d.ts

```diff
@@ -739,6 +739,11 @@ export interface InputProps extends TextInputProperties {
    */
   rightIconContainerStyle?: StyleProp<ViewStyle>;
 
+  /**
+   * Renders component in place of the React Native `TextInput` (optional)
+   */
+  inputComponent?: React.ComponentClass<any> | React.StatelessComponent<any>;
```

Comment:
Just use the component class

Reply:
Not proficient with typescript sor not sure I understand correctly, do you mean to use `React.Component`?

If my understanding is correct, `React.Component` is for a rendered component (eg. `<TextInput />`) while `React.ComponentClass` is for a component constructor (eg. `TextInput`) which is what we need here.

Thanks.

Reply:
Sorry, meant leave only `React.ComponentClass<any>`

Reply:
Is the a reason why we should forbid pure stateless components?

Reply:
Stateless components can't have refs, so methods we have on the `Input` component such as focus, blur etc will fail

Reply:
Did not know that, it's fixed, thanks.

Reply:
In fact I have a use-case for stateless components (I'm using them right now in my app), where I replace the TextInput with a `Touchable` component (that triggers a picker) to build a `Select` component:

```js
const SelectInputComponent = props => {
  const {onFocus, value, placeholder} = props;
  return (
    <TouchableWithoutFeedback onPress={onFocus}>
      <View style={{flex: 1, height: '100%'}}>
        <Text style={{paddingTop: 20, paddingLeft: 10, color: value ? 'black' : '#86939e'}}>
          {value || placeholder}
        </Text>
      </View>
    </TouchableWithoutFeedback>
  );
};
```

I think we should also allow these custom stateless components even if they don't work like a TextInput at all.

Reply:
What i'm saying is that if you use a stateless component you will get errors by we are assigning a ref to it.

Also is there any reason you changed the formatting on the whole file?